compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -509,6 +509,8 @@ trait Inferencing { this: Typer =>
         //     required: String
         //     val y: List[List[String]] = List(List(1))
         val hasUnreportedErrors = state.reporter.hasUnreportedErrors
```

Comment:
No need to define a local variable, it seems. We can simply return if `state.reporter.hasUnreportedErrors` is true.